pkg/restapi/rp/operation/operations.go

```diff
@@ -744,54 +747,131 @@ func (o *Operation) chapiResponseHandler(w http.ResponseWriter, r *http.Request)
 	// TODO save user Consent VC https://github.com/trustbloc/edge-adapter/issues/92
 	// TODO validate the user consent credential (expected rp and user DIDs, etc.)
 
-	customConsentVC, origConsentVC, err := parseWalletResponse(crCtx.PD, o.vdriReg, request.VerifiablePresentation)
-	if errors.Is(err, errInvalidCredential) {
-		handleError(w, http.StatusBadRequest, "malformed credentials")
+	local, remote, err := parseWalletResponse(crCtx.PD, o.vdriReg, request.VerifiablePresentation)
+	if err != nil {
+		if errors.Is(err, errInvalidCredential) {
+			handleError(w, http.StatusBadRequest, fmt.Sprintf("malformed credentials: %s", err.Error()))
+		} else {
+			handleError(w, http.StatusInternalServerError, fmt.Sprintf("failed to parse credentials : %s", err.Error()))
+		}
 
 		return
 	}
 
-	if err != nil {
-		handleError(w, http.StatusInternalServerError, fmt.Sprintf("failed to parse custom credentials : %s", err))
+	callback := make(chan *issuerResponseStatus)
 
-		return
+	for descriptorID, authz := range remote {
+		// TODO do not send multiple requests for each scope in an authz cred:
+		//  https://github.com/trustbloc/edge-adapter/issues/253
+		thid, err := o.requestRemoteCredential(authz, crCtx)
+		if err != nil {
+			if errors.Is(err, errInvalidCredential) {
+				handleError(w, http.StatusBadRequest, err.Error())
+			} else {
+				handleError(w, http.StatusInternalServerError, err.Error())
+			}
+
+			return
+		}
+
+		o.setWalletResponseContext(thid, &walletResponseCtx{
+			consentRequestCtx:    crCtx,
+			issuerResponseStatus: callback,
+			descriptorID:         descriptorID,
+		})
 	}
 
-	issuerDID, err := did.ParseDocument(customConsentVC.Subject.IssuerDIDDoc.Doc)
-	if err != nil {
-		msg := fmt.Sprintf("failed to parse did document : %s", err)
-		logger.Errorf(msg)
-		commhttp.WriteErrorResponse(w, http.StatusBadRequest, msg)
+	start := time.Now()
+	userData := make(map[string]*verifiable.Credential)
 
-		return
+	for id, cred := range local {
+		userData[id] = cred
 	}
 
-	// TODO Issuer's label on the connection record https://github.com/trustbloc/edge-adapter/issues/93
-	_, err = o.didClient.CreateConnection(crCtx.RPPairwiseDID, issuerDID)
-	if err != nil {
-		msg := fmt.Sprintf("failed to create didcomm connection : %s", err)
-		logger.Errorf(msg)
-		commhttp.WriteErrorResponse(w, http.StatusInternalServerError, msg)
+	// TODO the transient invitationData should be after deleted the CHAPI response is processed but the edge-core
+	//  Store API is missing a Delete() method: https://github.com/trustbloc/edge-core/issues/45
```

Comment:
https://github.com/trustbloc/edge-core/issues/45 has been implemented.

Reply:
@rolsonquadras Opened #255 and updated TODO